ACKategoriesCore/ACKUrl.swift

```diff
@@ -0,0 +1,25 @@
+//
+//  ACKUrl.swift
+//  ACKategories
+//
+//  Created by Alexandr Grigoryev on 21.02.2023.
+//
+
+import Foundation
+
+public struct ACKUrl: RawRepresentable {
+   public let url: URL
+
+   public init?(rawValue: String) {
+        guard let urlString = rawValue.addingPercentEncoding(withAllowedCharacters: .urlQueryAllowed),
```

Comment:
I am not sure, that this is correct, because if I have string `https://google.com/some%20thing` it will not work will it? Because the space character will be encoded twice, which is not correct.